docs/modules/ROOT/pages/references/decision-deployment-strategy.adoc

```diff
@@ -0,0 +1,160 @@
+If none exist that are suitable, a fork or own chart may be necessary.
++
+Helm charts usually deploy native `Deployments` or `StatefulSets` and are thus much easier to handle since they're not abstracted behind an additional CRD.
+However, a chart can still do changes that are effectively backwards-incompatible, requiring manual user interaction.
+One such example would be to rename the `PersistentVolumeClaim` template in a `StatefulSet`, which would require a sophisticated data migration.
+
+=== Breaking Change Scenario Example
+
+Let's consider the deployment strategy with a foreign PGO and the worst-case.
+Starting situation is:
+
+- Suppose PGO in v1 supports Postgresql up until version 14.
+- Suppose Postgresql v15 is released that requires user to upgrade their schemas.
+- PGO in v2 is released that supports Postgresql v15, but not v14 anymore.
+
+We have effectively 3 breaking changes:
+
+. PGO v2 API spec changed
+. Postgresql v15 requires data upgrade
+. PGO v2 doesn't support Postgresql v14
+
+This means:
+
+. We can't upgrade to Postgresql v15 without upgrading PGO to v2.
+. We can't upgrade to PGO v2 without having the possibility to run Postgresql v14 for the users that aren't ready for v15's data migration.
+
+The chance of such a constellation is slim, but not impossible and VSHN already experienced such scenarios.
+Even if Postgresql v15 doesn't require a user data migration, there needs to be an answer how to deal with upstream changes that result in some kind of impasse.
+
+== Proposals
+
+=== Proposal 1
+
+Operator:: in-place upgrades, handle API changes
+
+To deal with backwards-incompatible changes, each major version supported will run on a dedicated https://kb.vshn.ch/app-catalog/reference/glossary.html#_service_location[service cluster].
+This is to ensure that the old CRD version can still run on non-upgraded Postgresql instances so that old versions can be running until their EOL date.
+However, this massively increases the running costs.
+
+Whenever the user chooses the next major version, a new deployment is started in the service cluster that runs the new version, and an automatic data import is attempted from the old service cluster to the new one.
+
+This operator ({page-component-name}) abstracts and deploys a PGO resource and possibly glues it with additional resources, for example Prometheus rules.
+
+=== Proposal 2
+
+Helm:: in-place upgrades, handle API changes
+
+To deal with backward-incompatible changes, each major version supported will run on the last working chart version for that major version.
+This allows to skip the chart upgrade on old major versions, while the next major version will feature the changed Helm chart version.
+
+In the event that a critical change needs to be backported to an older major version, it should be easily possible to switch to a fork that contains the change on the running instances.
```

Comment:
Maybe a fork is not always necessary? We could evaluate the possibility to simply use a newer chart for an existing major version, IF there are no breaking changes.

Reply:
That's basically what I meant, but maybe it's not clear enough.
I intended to rollout newer chart versions for existing major versions, as long as it doesn't break.

However, I actually wanted to formulate the case when there ARE breaking changes.

Do you have suggestions for rephrasing?

Reply:
Yes, probably. Something like "As long as the there are no breaking changes, we'll update the charts as well" or something like that. It wasn't quite clear for me that the fork should only be the exception.

Reply:
reworded a bit